### ISSUE TITLE
functor/3 defective

### ISSUE THREAD

**Issue by flexoron** [trealla-prolog/trealla#558]
v1.28.14
```
?- functor(""(_),F,N).
   F = '[]nctor', N = 1.

and

?- functor("" (_),F,N).
   F = 0.
```

**Comment by UWN** [comment]
`functor("" (_),F,N).` should produce a syntax error since `[]` is not an operator. And since [Cor.2](http://www.complang.tuwien.ac.at/ulrich/iso-prolog/dtc2#Res_A78) the creation of such an operator is explicitly forbidden.

**Comment by UWN** [comment]
All this assumes
```
?- current_prolog_flag(double_quotes, atom).
   false.
```

**Comment by flexoron** [comment]
v1.28.19
```
?- functor([],F,N).
   F = [], N = 0.
?- functor("",F,N).
   F = [], N = 0.

?- functor([](_),F,N).
   F = [], N = 1.
?- functor(""(_),F,N).
Error: syntax error, near "[]", expected atom % Preceding examples suggest F = [], N = 1.
```

**Comment by infradig** [comment]
| ?- functor(""(_),F,N).

| Error: syntax error, near "[]", expected atom % Preceding examples
suggest F = [], N = 1.


The *near "[]"* thing is a bit confusing, but an error it should still be.

**Comment by flexoron** [comment]
```
hmm, ok, so 
functor([](_),F,N).
is an error, too?
```

**Comment by infradig** [comment]
No, [] is an atom.

**Comment by flexoron** [comment]
confused
?- atom("").
   true.

**Comment by flexoron** [comment]
Alright ...I guess the issue has been resolved.

**Comment by flexoron** [comment]
```
?- functor(1(_),F,N).
Error: syntax error, parsing number, line 1, ''
   false.

?- functor(1 (_),F,N).
   F = 0.
```

**Comment by flexoron** [comment]
v1.28.22
```
?- functor(* -,F,N).
Error: syntax error, incomplete, line 1 ''
   false.

Perhaps these having errors:

?- functor(+ -,F,N).
   F = (+), N = 1.

?- functor(* (-),F,N).
   false.

```

**Comment by UWN** [comment]
(please put different issues into different issues, this last point is about syntax only).
```
?- functor(+ -,F,N).
   F = (+), N = 1. % quite problematic 5.5.1 syntax extension. But possible.
?- functor(* (-),F,N).
   false. % unclear
?- T = (* (-)).
   T = (*). % very odd extension, better remove it
?- T = (* (-)), atom(T).
Error: syntax error, operator expected unfinished input '.', line 1, '' % even worse.
```


If you are looking into operators, you must take into account the operators defined. It seems that in Scryer you have loaded `library(debug)` for without it:

```
ulrich@p0:/opt/gupu/scryer-prolog$ target/release/scryer-prolog -f
?- functor(* (-),F,N).
   error(syntax_error(incomplete_reduction),read_term/3:1).
```

**Comment by flexoron** [comment]
My mistake, pardon me.

**Comment by flexoron** [comment]
v1.29.9
```
?- functor(0,F,F).
   F = 0.

?- functor(1,F,F).
   F = 0. % true and false
```

**Comment by flexoron** [comment]
v1.29.13
```
?- functor(1.0,F,A).
   F = 1.0, A = 0.

?- functor(1.0(_),F,A).
   F = 0.  % ?

```

**Comment by infradig** [comment]
?- X = 1.0(_).
Error: syntax error, operator expected unfinished input '.', line 1
   false.
?- X = f(1.0(_)).
   X = f(1.0).
?-

Don't know how that slipped through the parser, but i'll look into it.

**Comment by flexoron** [comment]
v1.29.14
```
?- functor(1.0.(_),F,A), B = b.
   false.  % What is wrong?
```

**Comment by flexoron** [comment]
Fixed with v1.29.15
